File: Golang/decorator/beverage.go
package decorator

type Beverage interface {
	Cost() float64
	GetDescription() string
}


type DarkRoast struct{}

func NewDarkRoast() Beverage {
	return &DarkRoast{}
} 

func (dr *DarkRoast) Cost() float64 {
	return 0.99
}

func (dr *DarkRoast) GetDescription() string {
	return "Dark Roast Coffee"
}

type Espresso struct{}

func NewEspresso() Beverage {
    return &Espresso{}
}

func (e *Espresso) Cost() float64 {
	return 1.99
}

func (e *Espresso) GetDescription() string {
	return "Espresso"
}

File: Python/decorator/beverage.py
from abc import ABC, abstractmethod


class Beverage(ABC):
    description: str = "Unknown Beverage"

    def getDescription(self) -> str:
        return self.description

    @abstractmethod
    def cost(self) -> float:
        raise NotImplementedError


class Espresso(Beverage):
    def __init__(self):
        self.description = "Expresso"

    def cost(self) -> float:
        return 1.99


class DarkRoast(Beverage):
    def __init__(self):
        self.description = "Dark Roast Coffee"

    def cost(self) -> float:
        return 0.99
